tests/0148-offset_fetch_commit_error_mock.c

```diff
@@ -0,0 +1,574 @@
+        uint64_t testid   = test_id_generate();
+        const int msgcnt  = 5;
+        rd_kafka_resp_err_t err;
+        rd_kafka_topic_partition_list_t *to_commit = NULL;
+
+        SUB_TEST_QUICK();
+
+        rd_kafka_mock_topic_create(mcluster, topic, 1, 1);
+
+        test_conf_init(&conf, NULL, 30);
+        test_conf_set(conf, "bootstrap.servers", bootstraps);
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "enable.auto.commit", "false");
+        test_conf_set(conf, "group.protocol", "consumer");
+
+        /* Seed the topic with messages */
+        test_produce_msgs_easy_v(topic, testid, 0, 0, msgcnt, 0,
+                                 "bootstrap.servers", bootstraps, NULL);
+
+        /* Consume same messages */
+        consumer = test_create_consumer(topic, NULL, conf, NULL);
+        test_consumer_subscribe(consumer, topic);
+        test_msgver_init(&mv, testid);
+        test_consumer_poll("receive first batch", consumer, testid, -1, 0,
+                           msgcnt, &mv);
+        test_msgver_verify("verify first batch", &mv, TEST_MSGVER_ALL, 0,
+                           msgcnt);
+        test_msgver_clear(&mv);
+
+        /* Set OffsetCommit errors */
+        rd_kafka_mock_push_request_errors(mcluster, RD_KAFKAP_OffsetCommit, 1,
+                                          expected_err);
+
+        if (variation ==
+            TEST_MANUAL_COMMIT_ERROR_VARIATION_STORE_OFFSET_MANUALLY) {
+                /* Variation 1: pass offsets to commit */
+                to_commit = rd_kafka_topic_partition_list_new(1);
+                rd_kafka_topic_partition_list_add(to_commit, topic, 0)->offset =
+                    msgcnt;
+        }
+
+        /* Sync commit */
+        err = rd_kafka_commit(consumer, to_commit, rd_false);
+        TEST_ASSERT(err == expected_err, "Expected error %s, got %s",
+                    rd_kafka_err2name(expected_err), rd_kafka_err2name(err));
+
+        /* Retry it, this time it should work */
+        err = rd_kafka_commit(consumer, to_commit, rd_false);
+        TEST_ASSERT(err == RD_KAFKA_RESP_ERR_NO_ERROR,
+                    "Expected error %s, got %s",
+                    rd_kafka_err2name(RD_KAFKA_RESP_ERR_NO_ERROR),
+                    rd_kafka_err2name(err));
+
+        RD_IF_FREE(to_commit, rd_kafka_topic_partition_list_destroy);
+
+        rd_kafka_destroy(consumer);
+
+        SUB_TEST_PASS();
+}
+
+typedef enum do_test_auto_commit_stale_member_epoch_error_variation_t {
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_NO_TIMEOUT = 0,
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_WITH_TIMEOUT =
+            1,
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_REVOKE_NO_TIMEOUT   = 2,
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_REVOKE_WITH_TIMEOUT = 3,
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION__CNT,
+} test_auto_commit_stale_member_epoch_error_variation_t;
```

Comment:
Same name here, they don't conflict as one is for the `enum` and the other is for the type
```suggestion
} do_test_OffsetCommit_automatic_stale_member_epoch_error_variation_t;
```